src/streamlit_app.py

```diff
@@ -227,8 +252,12 @@ async def amessage_iter() -> AsyncGenerator[ChatMessage, None]:
                     user_id=user_id,
                 )
                 messages.append(response)
-                st.chat_message("ai").write(response.content)
-            st.rerun()  # Clear stale containers
```

Comment:
The st.rerun() is still useful, right? I'd prefer to keep it, at least for the non-voice case.